packages/shared/src/components/feed/LeanFeed.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactElement } from 'react';
+import LeanFeedItemComponent from './LeanFeedItemComponent';
+import { useActiveFeedContext } from '../../contexts';
+import { FeedItem } from '../../hooks/useFeed';
+
+function LeanFeed({ items }: { items: FeedItem[] }): ReactElement {
+  const { feedRef } = useActiveFeedContext();
+
+  return (
+    <div className="grid grid-cols-2 gap-4" ref={feedRef}>
+      {items.map((item, i) => {
+        return <LeanFeedItemComponent key={i} item={item} />;
+      })}
```

Comment:
It could accept `children` as a function.  So we can call:

Render cards:
```jsx
<LeanFeed>
    {({ items }) => {
        return items.map((item, i) => {
            return <LeanFeedItemComponent key={i} item={item} />;
        }) 
    }}
</LeanFeed>
```

Add ad to the first place:
```jsx
<LeanFeed>
    {({ items }) => {
        return items.map((item, i) => {
            if (i === 0) {
                return (
                    <>
                        <Ad />
                        <LeanFeedItemComponent key={i} item={item} />;
                    </>
                )
            }
    
            return <LeanFeedItemComponent key={i} item={item} />;
        }) 
    }}
</LeanFeed>
```

Of course internal render prop function is in fact just a component:
```jsx
const FeedWithAds =  {({ items }) => {
    return items.map((item, i) => {
        return <LeanFeedItemComponent key={i} item={item} />;
    }) 
}}

<LeanFeed>
    {FeedWithAds}
</LeanFeed>
```

I think conceptually it would allow us to split view and business logic.